sentry/src/routes/campaign.rs

```diff
@@ -1,56 +1,286 @@
+    set_initial_remaining_for_campaign(&app.redis.clone(), campaign.id, campaign.budget).await.map_err(|_| ResponseError::BadRequest("Couldn't update remaining while creating campaign".to_string()))?;
+
+    // insert Campaign
+    match insert_campaign(&app.pool, &campaign).await {
+        Err(error) => {
+            error!(&app.logger, "{}", &error; "module" => "create_campaign");
+            match error {
+                PoolError::Backend(error) if error.code() == Some(&SqlState::UNIQUE_VIOLATION) => {
+                    Err(ResponseError::Conflict(
+                        "Campaign already exists".to_string(),
+                    ))
+                }
+                _ => Err(error_response),
+            }
+        }
+        Ok(false) => Err(ResponseError::BadRequest("Encountered error while creating Campaign; please try again".to_string())),
+        _ => Ok(()),
+    }?;
 
     Ok(success_response(serde_json::to_string(&campaign)?))
 }
 
+pub mod update_campaign {
+    use super::*;
+    use crate::fetch_campaign;
+
+    pub const CAMPAIGN_REMAINING_KEY: &'static str = "campaignRemaining";
+
+    pub async fn set_initial_remaining_for_campaign(redis: &MultiplexedConnection, id: CampaignId, amount: UnifiedNum) -> Result<bool, Error> {
+        let key = format!("{}:{}", CAMPAIGN_REMAINING_KEY, id);
+        redis::cmd("SETNX")
+            .arg(&key)
+            .arg(amount.to_u64())
+            .query_async(&mut redis.clone())
+            .await?;
+        Ok(true)
+    }
+
+    pub async fn increase_remaining_for_campaign(redis: &MultiplexedConnection, id: CampaignId, amount: UnifiedNum) -> Result<bool, Error> {
+        let key = format!("{}:{}", CAMPAIGN_REMAINING_KEY, id);
+        redis::cmd("INCRBY")
+            .arg(&key)
+            .arg(amount.to_u64())
+            .query_async(&mut redis.clone())
+            .await?;
+        Ok(true)
+    }
+
+    pub async fn decrease_remaining_for_campaign(redis: &MultiplexedConnection, id: CampaignId, amount: UnifiedNum) -> Option<UnifiedNum> {
+        let key = format!("{}:{}", CAMPAIGN_REMAINING_KEY, id);
+        let value = match redis::cmd("DECRBY")
+            .arg(&key)
+            .arg(amount.to_u64())
+            .query_async::<_, Option<i64>>(&mut redis.clone())
+            .await
+            {
+                Ok(Some(remaining)) => {
+                    // Can't be less than 0 due to max()
+                    Some(UnifiedNum::from_u64(max(0, remaining).try_into().unwrap()))
+                },
+                _ => None
+            };
+        value
+    }
+
+    pub async fn handle_route<A: Adapter>(
+        req: Request<Body>,
+        app: &Application<A>,
+    ) -> Result<Response<Body>, ResponseError> {
+        let campaign = req.extensions().get::<Campaign>().expect("We must have a campaign in extensions");
+        let modified_campaign = ModifyCampaign::from_campaign(campaign.clone());
```

Comment:
- [ ] `MofieidCampaign` should be created from the request body.
Later on we can `modify_campaign.apply(&mut campaign)` to apply the changes from the ModifiedCampaign => Campaign.